typescript/testsuite/api/api_service_client/api_client.ts

```diff
@@ -0,0 +1,181 @@
+            if (exception && exception.stack && exception.message) {
+                return err(exception as Error);
+            }
+            return err(new Error("Performing a post request threw an exception, but " +
+                "it's not an Error so we can't report any more information than this"));
+        }
+
+        return ok(null);
+    }
+
+    /*
+    Wait for healthy response
+    */
+    public async waitForHealthy(retries: number, retriesDelayMilliseconds: number): Promise<Result<null, Error>> {
+
+        const url: string = "http://"+this.ipAddr+":"+this.port+"/"+HEALTHCHECK_URL_SLUG;
+        let respResult: Result<axios.AxiosResponse<any>, Error> | null = null;
+
+        for (let i = 0 ; i < retries ; i++) {
+            try {
+                respResult = await this.makeHttpGetRequest(url);
+            } catch(exception: any) {
+                // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+                // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+                if (exception && exception.stack && exception.message) {
+                    return err(exception as Error);
+                }
+                return err(new Error("Making a HTTP get request threw an exception, but " +
+                    "it's not an Error so we can't report any more information than this"));
+            }
+            if (respResult.isOk()) {
+                break;
+            }
+            try {
+                await new Promise(resolve => setTimeout(resolve, retriesDelayMilliseconds));
+            } catch(exception: any) {
+                // Sadly, we have to do this because there's no great way to enforce the caught thing being an error
+                // See: https://stackoverflow.com/questions/30469261/checking-for-typeof-error-in-js
+                if (exception && exception.stack && exception.message) {
+                    return err(exception as Error);
+                }
+                return err(new Error("Creating a promise for the timeout threw an exception, but " +
+                    "it's not an Error so we can't report any more information than this"));
+            }
+        }
+
+        if (respResult === null) {
+            return err(new Error("Expected a response or error wrapped around Result, but got null instead. Ensure that retries is greater than 0."));
+        }
+        if (!respResult.isOk()){
+            return err(respResult.error);
+        }
+
+        const resp: axios.AxiosResponse<any> = respResult.value;
+        let bodyString: string = String(resp.data);
+
+        if (bodyString !== HEALTHY_VALUE) {
+            return err(new Error("Expected response body text '" + HEALTHY_VALUE + "' from endpoint '" + url + "' but got '" + bodyString +  "' instead"));
+        }
+
+        return ok(null);
+    }
+
+    public getPersonUrlForId(id: number): string {
```

Comment:
Helper functions should be `private`, and `static` if possible

Reply:
Can't make this static, only private ; since it depends on the properties of API Client